### ISSUE TITLE
Linear tolerance parameters are poorly named

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#11962]
## Rationale
`Executioner`'s linear tolerance parameter `l_abs_step_tol` is incorrectly named, and the parameter `l_tol` could be better named as `l_rel_tol`.

## Description
The parameter `l_abs_step_tol` actually gets used as an absolute tolerance, i.e., convergence is detected if the norm of the linear residual is less than this number. A *step* tolerance is compared to the change between iterates. Therefore, this parameter is just labeled wrong. It should be renamed to `l_abs_tol`. Similarly, internally, it appears the parameter "linear solver absolute step tolerance" that is set and gotten for `EquationsSystems` is mislabeled.

Furthermore, `l_tol` would be more descriptively named as `l_rel_tol`, and this would make the linear tolerances mirror the nonlinear tolerances, which is a better experience for the user.

## Impact
Parameter name changes.

**Comment by joshuahansel** [comment]
I don't know how these changes would be made, since `l_tol` is such a common parameter. Would deprecating it cause a lot of problems?

**Comment by friedmud** [comment]
Can you point to where `l_abs_step_tol` is doing the wrong thing?

`l_tol` is old - comes from libMesh/PETSc.  I don't mind deprecating it... but it will be a while before we remove it.

**Comment by joshuahansel** [comment]
Line numbers below correspond to MOOSE version `13a81b9d882a35e1877f5aed6691430cf9be4501`:
* `Executioner.C`:120: In constructor, `l_abs_step_tol` gets put in "linear solver absolute step tolerance" of `EquationSystems` object.
* `PetscSupport.C`:581: In `petscSetDefaultPCSide`, this is retrieved into local variable `atol`
* `PetscSupport.C`:590: In `petscSetDefaultPCSide`, local variable `atol` is used in `KSPSetTolerances` in the position corresponding to absolute tolerance.

You can verify this yourself with `-ksp_view`.

**Comment by YaqiWang** [comment]
Is this `l_abs_step_tol` ever used before Derek's last change in `PetscSupport.C:590`?

**Comment by joshuahansel** [comment]
You're right, it is used elsewhere: `FEProblemBase.C:5454`, where it appears to be used correctly. So if I'm understanding things correctly, it appears that it is only passed to PETSc incorrectly in `PetscSupport`.

**Comment by joshuahansel** [comment]
I just discovered a lot more wrong with `l_abs_step_tol`. See #11970.
